submissions/Shevchenko-26/sniffer/main.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func main() {
+	host := flag.String("host", "8.8.8.8", "required argument. Can be either domain name, like google.com or IP address like 172.217.3.110")
+	portsRange := flag.String("ports", "1-1000", "the range of ports to scan. Should be formatted as the following of <start_port>-<end_port>")
+	flag.Parse()
+
+	var start, end = 0, 1 << 16
+	var err error
+
+	if *portsRange != "" {
+		portsNums := strings.Split(*portsRange, "-")
+		if len(portsNums) != 2 {
+			log.Fatal("Ports range is invalid. Should be in form of <start_port>-<end_port>")
+		}
+		start, err = strconv.Atoi(portsNums[0])
+		if err != nil {
+			log.Fatal("Starting port was specified incorrectly. Aborting...")
+		}
+		endPort, err := strconv.Atoi(portsNums[1])
+		if err != nil {
+			log.Fatal("Second port was specified incorrectly. Aborting...")
+		}
+		if endPort > end {
+			log.Fatal("Second port overflows maximum value of 65536. Aborting...")
+		}
+		end = endPort
+	}
+
+	var ports []int
+	for i := start; i < end; i++ {
+
+		_, err := net.DialTimeout("tcp",
+			fmt.Sprintf("%s:%v", *host, i), time.Millisecond*300)
+		if err != nil {
+			if err, ok := err.(*net.OpError); ok {
+				if !err.Err.(*net.DNSError).IsTimeout {
```

Comment:
No need to create dedicated check for this error. If you want to check for host validity maybe its better to dns lookup host before dial (in separate function)?:
```go
dns := net.Resolver{}
ctx := context.WithTimeout(context.Background(), time.Millisecond * 300)
_, err := dns.LookupHost(ctx, host)
if err != nil {
     log.Fatal("Wrong host")
}
```
Or keep it as it is but assert `err.Err.(*net.DNSError). IsTimeout ` more  safely.
```go
if err, ok := err.Err.(*net.DNSError); ok && err.IsTimeout {
}
```

Reply:
Or try `errors.As` from go `1.13`